### PR TITLE
Smart Notifier

### PR DESCRIPTION
**Pull request by michaloo**
# 0.13.0
* introduces `smartNotifierMiddleware` and `smartNotifierHandler` utils
* introduces `ctx.smartNotifierResponse` context object to build flow response
* added `skipSignatureValidation` `Hull.Connector` options to skip smartNotifier signature validation
* **breaking**: removed `serviceMiddleware` and `service` param on `Hull.Connector`, new way of applying service related objects to the Context Object is the `use` method on the Connector instance and initialize the service objects directly on the `req.hull.service` namespace
* restructurized tests folders and files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,5 +103,10 @@
       "spy",
       "expect"
     ]
+  },
+  "nyc": {
```
Comment (sbellity) [inline package.json]: @michaloo what is this ?

File: README.md [hunk 2]
```diff
@@ -654,6 +654,50 @@ app.use('/notify', handler);
...
+import { smartNotificationHandler } from "hull/lib/utils";
+const app = express();
+
+const handler = smartNotificationHandler({
+  handlers: {
+    "ship:update": function(ctx, messages = []) {},
+    "segment:update": function(ctx, messages = []) {},
+    "segment:delete": function(ctx, messages = []) {},
+    "user:update": function(ctx, messages = []) {
+      console.log("Event Handler here", ctx, messages);
```
Comment (sbellity) [inline README.md]: Need an example of response with flow control

File: package.json (hunk 1 above)
Comment (michaloo) [inline package.json]: Code coverage tool I'm testing - istanbul.
I've also tested on one of my open-source projects integration of CircleCI with CodeCov which is tool for presenting the coverage information:
https://codecov.io/gh/michaloo

File: README.md (hunk 2 above)
Comment (michaloo) [reply README.md]: Added it here and more documentation below.

**Pull request merged**